### ISSUE TITLE
Adding new server via provider ssh Key

### ISSUE THREAD

**Issue by iantearle** [vitodeploy/vito#272]
Upon creating a server on DigitalOcean I got a grey modal pop up with Error 500. 
Subsequent attempts at creating a server result in a red message "Provider Error: DigitalOcean SSH Key".

My digital ocean account appears to have an unfinished droplet created. 

I can't seem to ssh on as Vito user either. I can using my ssh keys onto as root.

**Comment by saeedvaziry** [comment]
You need to delete the existing key on your DO dashboard

**Comment by erenilhan** [comment]
I have the same issue :/ I'm tried of everything but no solution yet.

**Comment by saeedvaziry** [comment]
Will try to reproduce it

**Comment by saeedvaziry** [comment]
Still looks like working to me,
@erenilhan @paligiannis @iantearle can you you write the exact steps to reproduce?

**Comment by erenilhan** [comment]
I deleted all keys and ssh from Digitalocean.

<img width="1290" alt="Screenshot 2024-08-16 at 10 27 10 PM" src="https://github.com/user-attachments/assets/b8e4d95a-86a0-42f8-ad26-1167c92a675e">

<img width="778" alt="Screenshot 2024-08-16 at 10 27 28 PM" src="https://github.com/user-attachments/assets/00246c5b-c6d5-4cb5-975d-161cd1c35720">

<img width="1225" alt="Screenshot 2024-08-16 at 10 27 40 PM" src="https://github.com/user-attachments/assets/f5ded88d-a25d-4072-8edc-1f03b3b261b4">

<img width="1097" alt="Screenshot 2024-08-16 at 10 28 15 PM" src="https://github.com/user-attachments/assets/e881e427-539d-43c0-9c78-73d531701c0a">

<img width="1463" alt="Screenshot 2024-08-16 at 10 30 29 PM" src="https://github.com/user-attachments/assets/295e47ca-c861-4279-bc59-2d09841d0916">

First click
<img width="1151" alt="Screenshot 2024-08-16 at 10 30 36 PM" src="https://github.com/user-
attachments/assets/bfa86527-13d1-4e40-b1e1-372cf74a9661">

Second click
<img width="1220" alt="Screenshot 2024-08-16 at 10 30 55 PM" src="https://github.com/user-attachments/assets/8914ec2c-e240-45a7-af7d-b1ec6223f775">

but ssh is here

<img width="740" alt="Screenshot 2024-08-16 at 10 33 57 PM" src="https://github.com/user-attachments/assets/1c276a98-5820-498f-97df-00f099b8ba56">

**Comment by saeedvaziry** [comment]
@erenilhan the "First click" screenshot is unavailable and I can't see.

The second error makes sense because the key already exists due to first attempt error.

**Comment by erenilhan** [comment]
Ah, sorry, the first click was about the server error.
First click

![image](https://github.com/user-attachments/assets/2f289d7c-94f2-4cd6-9b36-1370d13f2401)

@saeedvaziry

**Comment by saeedvaziry** [comment]
@erenilhan can you create a server on DO manually? From DO UI

**Comment by erenilhan** [comment]
Yes, I tried and ita created a new one

**Comment by paligiannis** [comment]
@erenilhan I have exactly the same issues. I tried also from 2 different accounts with the same results. Also, adding SSH to DO didn't make a difference. Also, no other logs produced. Only 'Provider Error: DigitalOcean SSH Key'. Is there any other way to troubleshoot?

**Comment by erenilhan** [comment]
still the same for me, I will reinstall and debug locally. it works for Hertzner but not for DO.
@paligiannis

**Comment by paligiannis** [comment]
@erenilhan I tried with Hetzner, works perfectly and fast! Why to use DO? Hetzner also seems to be cheaper and faster (my opinion) I am also sticked to DO but I need alternatives. Now running about 250-300 servers using K8S and Hetzner is a good alternative.

**Comment by iantearle** [comment]
@saeedvaziry I just spun up a new server and installation. Started a fresh, still the same errors as screenshotted above. I also tried adding an existing key in DO to Vito. 
When I do so I get this:
![Screenshot 2024-08-20 at 14 38 13](https://github.com/user-attachments/assets/8ae54169-4aaf-44a5-a7ea-1f5533ac5b30)

**Comment by iantearle** [comment]
Equally looking through the security logs droplet.create is immediately followed by a droplet.destroy. Is it self destroying on any error?

**Comment by saeedvaziry** [comment]
@iantearle if vito deosn't get the expected response codes it will consider a failure and will try to drop the droplet to avoid costs.

my guess is the plan maybe doesn't exist. will try again to reproduce.

also @iantearle, on the 500 error you got, you should be able to see logs on laravel.log file.

**Comment by iantearle** [comment]
@saeedvaziry that's exactly where I was heading: two notable lines:
```
[2024-08-20 13:33:08] production.ERROR: Failed to create server on DigitalOcean {"id":"unprocessable_entity","message":"The image you selected is no longer available"} 
[2024-08-20 13:37:59] production.ERROR: Undefined array key "8.2" {"userId":1,"exception":"[object] (ErrorException(code: 0): Undefined array key \"8.2\" at /home/vito/vito/app/Models/Service.php:52)
```
Let me know if you want a full dump of the logs #64 lines.

**Comment by erenilhan** [comment]
@iantearle  I also got this message yesterday, but when I changed the image and the version of ubuntu it worked. but interestingly I never got the error (image error) that I got after creating the server.

**Comment by saeedvaziry** [comment]
I thought the image ids issue has been resolved in #271 

I don't think it is related to the PHP version tho

**Comment by saeedvaziry** [comment]
Alright now I see :)

Although the PR merged but we haven't released the new version of Vito yet.

Soon will release the next version to fix all these.

**Comment by saeedvaziry** [comment]
I'd raise a separate issue for the php 8.2 thingy. Hard to follow

**Comment by saeedvaziry** [comment]
Update to 1.10.0, the problem should be gone!

**Comment by paligiannis** [comment]
Thanks for the 1.10.0. Basically I'm unable to update to the latest 1.10. I run the command update and I get the following: Vito updated successfully to 1.9.1! 🎉 but latest is 1.10. 
Any troubleshooting?
